sample-graphs/update.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+set -eu
+
+SOURCECRED_REMOTE="git@github.com:sourcecred/sourcecred.git"
+
+# Taken from https://stackoverflow.com/a/246128
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+# Use a clean SOURCECRED_DIRECTROY to avoid cache pollution.
+# Maybe reconsider this if we start storing really large projects here
+# (but putting really large projects' graphs in this Git repository is likely
+# a bad idea anyway).
+export SOURCECRED_DIRECTORY="$(mktemp -d --suffix ".sourcecred-directory")"
+
+# Just clone a fresh copy of SourceCred for convenience.
+sourcecred_repo="$(mktemp -d --suffix ".sourcecred-repo")"
+git clone "${SOURCECRED_REMOTE}" "${sourcecred_repo}"
+cd "${sourcecred_repo}"
```

Comment:
Suggestion: Rather than `cd`ing around in the main script, only change
directories within a subshell from which you actually invoke SourceCred:

```sh
sourcecred() {
  (cd "${sourcecred_repo}"; node ./bin/sourcecred.js "$@")
}

output_dir="$(dirname "$0")"

load_repo() {
  sourcecred load "$1"/"$2"
  sourcecred export-graph "$1"/"$2" >"${output_dir}/$1_$2.json"
}
```

Note the subshell `( … )`, which scopes the `cd`. The redirection
happens _outside_ the subshell and so is easier to think about.

The tokens `"$@"` mean “all the arguments to this function, individually
quoted in the correct, sane way that you would expect”.

Reply:
Thanks! This is a lot cleaner.